api/provider/aws/test_aws/deploy_list_test.go

```diff
@@ -0,0 +1,138 @@
+package test_aws
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/golang/mock/gomock"
+	provider "github.com/quintilesims/layer0/api/provider/aws"
+	"github.com/quintilesims/layer0/api/tag"
+	awsc "github.com/quintilesims/layer0/common/aws"
+	"github.com/quintilesims/layer0/common/config/mock_config"
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDeployList(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+
+	tags := models.Tags{
+		{
+			EntityID:   "dpl_id1",
+			EntityType: "deploy",
+			Key:        "name",
+			Value:      "dpl1",
+		},
+		{
+			EntityID:   "dpl_id1",
+			EntityType: "deploy",
+			Key:        "version",
+			Value:      "1",
+		},
+		{
+			EntityID:   "dpl_id1",
+			EntityType: "deploy",
+			Key:        "arn",
+			Value:      "arn:aws:ecs:region:012345678910:task-definition/l0-test-dpl:1",
```

Comment:
Compare the ARNs in these tags to the ARNs that you generate below in [`listTaskDefinitionPagesfn()`](https://github.com/quintilesims/layer0/pull/418/files#diff-a3d4f7c16b15c14bcb92929c3a14dba0R97)

`"arn:aws:ecs:region:012345678910:task-definition/l0-test-dpl:1"`
vs
`"arn:aws:ecs:region:012345678910:task-definition/l0-test-dpl1:1"`